text/dotnet-core/rfcs/0004-extend-dotnet-sdk.md

```diff
@@ -0,0 +1,50 @@
+# Extend .Net SDK Dependency
+
+## Proposal
+
+Reconfigure the .Net SDK dependency to be identical to the one that can be downloaded from Microsoft which includes the runtimes that the SDK is dependent on to run.
+
+## Motivation
+As of .Net SDK version ~6.0.100 publish now longer properly functions for certain types of .Net apps. Upon testing we were able to get the buildpack to function again if we used an .Net SDK dependency that still had the runtimes libraries packaged with it. After further investigation we discovered that the removal of those local runtimes was what was causing the failure, leading us to believe that the .Net SDK no longer follows the runtime search path that it used to.
+
+It is becoming ever more apparent to me that Microsoft is dead set on not having flexibility of file location during `dotnet publish` therefore I think that we should capitulate to Microsoft and create an expected .Net Hive installation. I think that this will make the buildpack less brittle to change in the future and will construct a build environment that is more similar to what .Net developers actually have installed on their machines and what Microsoft has in their Docker images.
+
+There are some additional advantages to using the .Net SDK dependency with its included runtimes chief among them being that newer versions of the .Net SDK can compile any version of an app for any given version of a runtime that is in the same `major.minor` family. This would mean that we would only have to support one version of the .Net SDK for every minor version as opposed to the two we currently support. On top of that we would no longer need to align our launch time runtime versions one-to-one with a single .Net SDK meaning that our version resolution logic would be vastly simplified. Also because we would only need to support one .Net SDK, users that are building offline capable buildpacks would be able to build a significantly smaller offline buildpack. Below are tables of each of the sizes of each dependency and the size of the overall offline bundle right now versus the proposed state.
```

Comment:
`
```suggestion
There are some additional advantages to using the .Net SDK dependency with its included runtimes chief among them being that newer versions of the .Net SDK can compile any version of an app for any given version of a runtime that is in the same `major.minor` family. This would mean that we would only have to support one version of the .Net SDK for every major version as opposed to the two we currently support. On top of that we would no longer need to align our launch time runtime versions one-to-one with a single .Net SDK meaning that our version resolution logic would be vastly simplified. Also because we would only need to support one .Net SDK, users that are building offline capable buildpacks would be able to build a significantly smaller offline buildpack. Below are tables of each of the sizes of each dependency and the size of the overall offline bundle right now versus the proposed state.
```

right?